### ISSUE TITLE
ts-expect-error suppresses wrong error

### ISSUE THREAD

**Issue by DetachHead** [microsoft/TypeScript#43715]
# Bug Report

### 🔎 Search Terms
ts-ignore ts-expect-error suppresses wrong error

### 🕗 Version & Regression Information
4.3.0-dev-20210415

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210415#code/C4TwDgpgBAYg9nAPAZWAJyhAHsCA7AEwGcoj0BLPAcwBooA5AVwFtMd9io8WAjCNAHxQAvFwgA3fgFgAULIIQAxgBsAhmmiK4eMlABmCAFywEiWVAtQA5AbhWa5y1Z7qrUAPTutaDYuDKQTB84NCJZAVl5JTUNKC0dYCgXNGN4JEcLAEY6T29ff0D+NBCwmUtrZLdPAjgIIjwrROAAC2KAdzowDSIWVR4ApKVVRiJoFv5oNtUSVWUNVQJA1Twg4owiOChQSDjmpQBrfigCcgIGxJ44cYxFPcV9yiotvahu4HDImQUVdU1tXRcAC9UqYMh53AABYBEAC02EgfjhwTQYOyYOcrnBeE2RRCdAk+GecEYVGazwmUCmMxWuIwfTgkmeqkSVNIjDAXTqowIHzkMiAA)

### 💻 Code

```ts
type Foo<Str extends string, Num extends number> = never

declare const foo: Foo<
    'foo',
    'bar' //correctly errors
>

declare const bar: Foo<
    1, //correctly errors
    'bar' //doesn't throw, presumably because there was already an error so type checker didn't bother checking the rest
>

declare const baz: Foo<
    //@ts-expect-error
    1,
    'bar' //no error, even though there was an error above that was suppressed
>
```

### 🙁 Actual behavior
no error on incorrect type that's not suppressed

### 🙂 Expected behavior
compiler error on incorrect type that isn't being suppressed by `ts-expect-error`

**Comment by RyanCavanaugh** [comment]
ts-expect-error doesn't change what errors get issued and what don't. It's more akin to a text-based filter applied at the very end of the process.
